### PR TITLE
Add cluster capabilities in spec

### PR DESCRIPTION
**Pull request by mrajashree**
Issue: https://github.com/rancher/rancher/issues/18275

### CODE REVIEW

File: apis/management.cattle.io/v3/cluster_types.go [hunk 1]
```diff
@@ -215,9 +216,9 @@ type ImportYamlOutput struct {
 
 type Capabilities struct {
 	LoadBalancerCapabilities LoadBalancerCapabilities `json:"loadBalancerCapabilities,omitempty"`
-	IngressCapabilities      []IngressCapabilities    `json:"ingressCapabilities,omitempty"`
-	NodePoolScalingSupported bool                     `json:"nodePoolScalingSupported,omitempty"`
-	NodePortRange            string                   `json:"nodePortRange,omitempty"`
+	IngressCapabilities      []IngressCapabilities    `json:"ingressCapabilities,omitempty" norman:"noupdate"`
+	NodePoolScalingSupported bool                     `json:"nodePoolScalingSupported,omitempty" norman:"noupdate"`
+	NodePortRange            string                   `json:"nodePortRange,omitempty" norman:"noupdate"`
 }
```
Comment (prachidamle) [inline apis/management.cattle.io/v3/cluster_types.go]: noupdate will disallow users from updating the other capabilities. But we probably need to disallow users from setting them during cluster creation as well?

Comment (mrajashree) [reply apis/management.cattle.io/v3/cluster_types.go]: Added the nocreate tag

**Comment by prachidamle** [comment]
LGTM

**Comment by mrajashree** [comment]
Closing as per https://github.com/rancher/rancher/issues/18275#issuecomment-490601105

**Pull request closed without merging**